sycl/include/sycl/vector_preview.hpp

```diff
@@ -107,25 +107,6 @@ template <typename T> class GetOp {
   }
 };
 
-// Forward declarations
-template <typename TransformedArgType, int Dims, typename KernelType>
```

Comment:
Wrong question :) You comment implies that these shouldn't be here because you tried to remove them as well. As to some issues - I'm not aware of any but waiting for the CI to finish.

Reply:
Yes, they shouldn't be here as they are aren't strictly related to `sycl::vec`.
Here's my theory:
I guess when I splitted `types.hpp` to `vector.hpp` and `vector_preview.hpp`, I incorrectly included these `RoundedRangeKernel` declarations in the vector headers (even `vector.hpp` has these: https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/vector.hpp#L185), thus creating a dependency between vector headers and device copyable code. That's why I was getting an error when trying to remove these from `vector_preview`.

But, when you created the `is_device_copyable` header, you added these declarations to it (https://github.com/intel/llvm/pull/13755/files) and removed the dependency. And so, now these declarations can be removed from `vector_preview.hpp`.